### PR TITLE
SDK must work even if there are no extensions

### PR DESCRIPTION
**Pull request by hectorhdzg**
Allowing core to work even if extensions are not present, most basic scenario of SDK core+ channel

### CODE REVIEW

File: vNext/shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ export class AppInsightsCore implements IAppInsightsCore {
         this.logger = new DiagnosticLogger(config);
 
         // Initial validation
-        extensions.forEach((extension: ITelemetryPlugin) => {
+        this._extensions.forEach((extension: ITelemetryPlugin) => {
```
Comment (markwolff) [inline vNext/shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts]: I believe `this._extensions` is an empty array at this point

File: vNext/shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts [hunk 2]
```diff
@@ -145,11 +145,6 @@ export class AppInsightsCore implements IAppInsightsCore {
             }
         });
 
-        // Remove sender channels from main list
-        if (c < this._extensions.length) {
-            this._extensions.splice(c);
-        }
```
Comment (markwolff) [inline vNext/shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts]: What was this code meant to do?

File: vNext/shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts (hunk 1 above)
Comment (hectorhdzg) [reply vNext/shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts]: I see, so is actually checking the input, my mistake reverting this change

File: vNext/shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts (hunk 2 above)
Comment (hectorhdzg) [reply vNext/shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts]: This code is basically removing ChannelController from the array, I don't see any use of this, _extensions is only used in track method after this and having Channel Controller in there doesn't look to affect any scenario, _extensions is also a public property but is only used in SKU to identify if Properties extension is available

File: vNext/shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts (hunk 1 above)
Comment (hectorhdzg) [reply vNext/shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts]: Merged both extensions and config.extension so validation applies to everything

File: vNext/shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts (hunk 2 above)
Comment (hectorhdzg) [reply vNext/shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts]: Adding this logic back and including check for empty extensions during track

File: vNext/shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts [hunk 3]
```diff
@@ -238,114 +235,23 @@ export class AppInsightsCore implements IAppInsightsCore {
...
 
         if (CoreUtils.isNullOrUndefined(telemetryItem.iKey)) {
-            this._notifiyInvalidEvent(telemetryItem);
+            this._notififyInvalidEvent(telemetryItem);
             throw Error("telemetry instrumentationKey required");
         }
     }
 
-    private _notifiyInvalidEvent(telemetryItem: ITelemetryItem): void {
+    private _notififyInvalidEvent(telemetryItem: ITelemetryItem): void {
```
Comment (markwolff) [inline vNext/shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts]: typo

**Pull request merged**